sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1683,6 +1684,7 @@ pi_result ExecCGCommand::SetKernelParamsAndLaunch(
     case kernel_param_kind_t::kind_pointer: {
       Plugin.call<PiApiKind::piextKernelSetArgPointer>(Kernel, Arg.MIndex,
                                                        Arg.MSize, Arg.MPtr);
+      IndirectAccessesNeeded = true;
```

Comment:
Can't we do even better than this? Can't we track in the kernel somehow that some other memory is accessed through this pointer indirectly? E.g. if it is a pointer to an integer, then we don't need to set this attribute, but only if it is a pointer to memory where another pointer is contained *and it is actually used*.

Reply:
@smaslov-intel - It's not about indirect loads through pointers.

"Indirect here" is more descended from OpenCL SVM.  OpenCL SVM required the kernel to state *every* pointer it *might* touch inside a kernel.  "Indirect" here says "you can just use them without telling me everything you might use"